packages/trpc/server/routers/viewer/availability.tsx

```diff
@@ -131,11 +131,13 @@ export const availabilityRouter = createProtectedRouter()
           })),
         },
       };
-
+      
+      const hasDefaultScheduleId = await hasDefaultSchedule(user, prisma);
```

Comment:
I might be confused but isn't the code behaving the same way unless we check for `hasDefaultScheduleId` to prevent `prisma.schedule.create`

Reply:
The logic is right. But in `hasDefaultSchedule` function, it will also check if `schedule` db has a record.
```js
const hasDefaultSchedule = async (user: Partial<User>, prisma: PrismaClient) => {
  const defaultSchedule = await prisma.schedule.findFirst({
    where: {
      userId: user.id,
    },
  });
  return !!user.defaultScheduleId || !!defaultSchedule;
};
```
If `prisma.schedule.create` run before this check, user will never get a `defaultScheduleId`.

Or I can just skip the schedule db check in `hasDefaultSchedule` function?

Reply:
cc @emrysal